autopush/websocket.py

```diff
@@ -773,9 +773,14 @@ def _check_message_table_rotation(self, previous):
         self.transport.pauseProducing()
         # Check for table rotation
         cur_month = previous.get("current_month")
+        # Previous month user or new user, flag for message rotation and
+        # set the message_month to the router month
         if cur_month != self.ps.message_month:
-            # Previous month user or new user, flag for message rotation and
-            # set the message_month to the router month
+            if cur_month not in self.ps.settings.message_tables.keys():
```

Comment:
I figured that the most common case would be that the previous "current_month" would equal the current month (as most clients would reconnect in the same month) thus saving the lookup for multiple values. Less likely would be: The connection is from last month, the connection is from a month that expired, the connection is one that was already forwarded (the tomorrow is another month addendum). If this is not the case, or if it's the same cost or cheaper to compare a value against a list than to compare against a single, I'm fine with creating an elif.